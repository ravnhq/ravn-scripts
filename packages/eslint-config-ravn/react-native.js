module.exports = {
  extends: ['./react.js'],
  plugins: ['react-native'],
  env: {
    'react-native/react-native': true,
  },
  rules: {
    'no-console': [
      'error',
      { allow: ['info', 'warn', 'error', 'ignoredYellowBox'] },
    ],

    'react-native/no-unused-styles': 'error',
    'react-native/no-raw-text': [
      'error',
      {
        // Should be set in the project config
        skip: [],
      },
    ],
  },
}
