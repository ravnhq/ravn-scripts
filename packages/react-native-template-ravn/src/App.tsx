import React, {useState, useCallback, useEffect} from 'react';
import {createAppContainer} from 'react-navigation';
import * as RNLocalize from 'react-native-localize';
import AppNavigator from './screens/MainStackNavigator';
import {ThemeProvider} from 'src/ui/styled-components';
import luxulTheme from 'src/ui/theme';
import {setI18nConfig} from './i18n';

const AppContainer = createAppContainer(AppNavigator);

const useForceUpdate = () => {
  const [, updateState] = useState({});
  const forceUpdate = useCallback(() => updateState({}), []);
  return forceUpdate;
};

export default function App() {
  setI18nConfig();

  const forceUpdate = useForceUpdate();

  useEffect(() => {
    const handleLocalizationChange = () => {
      setI18nConfig();
      forceUpdate();
    };

    RNLocalize.addEventListener('change', handleLocalizationChange);

    return () => {
      RNLocalize.removeEventListener('change', handleLocalizationChange);
    };
  });

  const [theme] = React.useState(luxulTheme);

  return (
    <CurrentRouterContextProvider>
      <ThemeProvider theme={luxulTheme}>
        <AppContainer screenProps={{theme}} />
      </ThemeProvider>
    </CurrentRouterContextProvider>
  );
}
