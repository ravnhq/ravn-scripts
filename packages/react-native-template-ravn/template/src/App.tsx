import React, {useState, useCallback, useEffect} from 'react';
import {createAppContainer} from 'react-navigation';
import * as RNLocalize from 'react-native-localize';
import {ApolloProvider} from '@apollo/react-hooks';
import MainStack from './screens/MainStack';
import {ThemeProvider} from '@styled-components';
import theme from 'src/ui/theme';
import {setI18nConfig} from './i18n';
import client from 'src/graphql/client';

const AppContainer = createAppContainer(MainStack);

const useForceUpdate = () => {
  const [, updateState] = useState({});
  const forceUpdate = useCallback(() => updateState({}), []);
  return forceUpdate;
};

export default function App() {
  setI18nConfig();

  const forceUpdate = useForceUpdate();

  useEffect(() => {
    const handleLocalizationChange = () => {
      setI18nConfig();
      forceUpdate();
    };

    RNLocalize.addEventListener('change', handleLocalizationChange);

    return () => {
      RNLocalize.removeEventListener('change', handleLocalizationChange);
    };
  });

  // This is a piece of state so we have the option to change it
  const [selectedTheme] = React.useState(theme);

  return (
    <ApolloProvider client={client}>
      <ThemeProvider theme={theme}>
        <AppContainer screenProps={{theme: selectedTheme}} />
      </ThemeProvider>
    </ApolloProvider>
  );
}
