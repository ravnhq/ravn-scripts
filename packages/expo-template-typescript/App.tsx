/**
 * This is the base or root of the entire react native expo app. The entry point. The beginning and the end. Alpha and Omega. App.tsx.
 */

import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Open up App.tsx to start working on your app!</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
})
