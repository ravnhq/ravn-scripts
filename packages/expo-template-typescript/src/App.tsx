import React from 'react'
import styled from 'styled-components/native'
import i18n from 'i18n-js'
import { StatusBar } from 'react-native'

const Container = styled.View`
  flex: 1;
  background-color: #1c2225;
  justify-content: center;
  align-items: center;
`

const Title = styled.Text`
  font-size: 20px;
  font-weight: 500;
  color: palevioletred;
`

export default function App() {
  return (
    <Container>
      <StatusBar barStyle="light-content" />
      <Title>{i18n.t('BROLOLO')}</Title>
    </Container>
  )
}
