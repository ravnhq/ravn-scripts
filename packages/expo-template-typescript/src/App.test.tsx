import React from 'react'
import { render } from '@testing-library/react-native'
import App from './App'

describe('App', () => {
  it('renders', () => {
    const { asJSON } = render(<App />)
    expect(asJSON()).toMatchSnapshot()
  })
})
