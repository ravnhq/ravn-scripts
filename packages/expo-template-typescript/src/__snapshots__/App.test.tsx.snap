// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1c2225",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "palevioletred",
            "fontSize": 20,
            "fontWeight": "500",
          },
        ]
      }
    >
      [missing "en.BROLOLO" translation]
    </Text>
  </View>
</View>
`;
